### PR TITLE
EmptyReference exception along with unit tests

### PR DESCRIPTION
**Pull request by sonofmun**
Fix for #169.
I am not sure that I chose the best way to detect the empty references (https://github.com/Capitains/MyCapytain/compare/empty_reference?expand=1#diff-74f2c42fe77e1484362851899deae795R251) (I hope that points to the correct line. In any case, it is line 251 in `MyCapytain/resources/texts/local/capitains/cts.py`).

### CODE REVIEW

**Comment by PonteIneptique** [comment]
Hey @sonofmun :D 
Currently in the us and on my way to some breakfast.
I think if you want this to be available for Hooktest quickly, the PR should focus master and not dev (dev is the current branch for DTS and some other fixes)
I have seen some printing remaining and other things I have some doubt about, I'll add this in the review.

File: MyCapytain/resources/texts/local/capitains/cts.py [hunk 1]
```diff
@@ -248,6 +248,11 @@ def getValidReff(self, level=None, reference=None, _debug=False):
                 message = ", ".join(duplicates)
                 warnings.warn(message, DuplicateReference)
             del duplicates
+            empties = [n for n in passages if n.strip('.') != n or n == '']
+            print(passages)
```
Comment (PonteIneptique) [inline MyCapytain/resources/texts/local/capitains/cts.py]: Probably something that made it to the final commit that should not have ? :)

File: tests/testing_data/texts/duplicate_references.xml [hunk 2]
```diff
@@ -69,24 +69,19 @@
                 <div type="textpart" subtype="book" n="1">
                     <div type="textpart" subtype="poem" n="pr">
                         <l n="1">Spero me secutum in libellis meis tale temperamen-</l>
-                        <l n="1">tum, ut de illis queri non possit quisquis de se bene </l>
```
Comment (PonteIneptique) [inline tests/testing_data/texts/duplicate_references.xml]: Why are these line removed ?

File: tests/testing_data/texts/duplicate_references.xml [hunk 3]
```diff
@@ -69,24 +69,19 @@
...
                     </div>
                 </div>
                 <div type="textpart" subtype="book" n="3">
                     <div type="textpart" subtype="poem" n="1">
                         <l n="1">Spero me secutum in libellis meis tale temperamen-</l>
                     </div>
-                    <div type="textpart" subtype="poem" n="1">
-                        <l n="2">Spero me secutum in libellis meis tale temperamen-</l>
```
Comment (PonteIneptique) [inline tests/testing_data/texts/duplicate_references.xml]: Why are these line removed ?

File: tests/testing_data/texts/duplicate_references.xml [hunk 4]
```diff
@@ -69,24 +69,19 @@
                 <div type="textpart" subtype="book" n="1">
                     <div type="textpart" subtype="poem" n="pr">
                         <l n="1">Spero me secutum in libellis meis tale temperamen-</l>
-                        <l n="1">tum, ut de illis queri non possit quisquis de se bene </l>
                     </div>
                 </div>
                 <div type="textpart" subtype="book" n="1">
                     <div type="textpart" subtype="poem" n="2">
                         <l n="1">Spero me secutum in libellis meis tale temperamen-</l>
-                        <l n="1">tum, ut de illis queri non possit quisquis de se bene </l>
```
Comment (PonteIneptique) [inline tests/testing_data/texts/duplicate_references.xml]: Why are these line removed ?

File: MyCapytain/resources/texts/local/capitains/cts.py (hunk 1 above)
Comment (sonofmun) [reply MyCapytain/resources/texts/local/capitains/cts.py]: I will remove that.

File: tests/testing_data/texts/duplicate_references.xml (hunk 4 above)
Comment (sonofmun) [reply tests/testing_data/texts/duplicate_references.xml]: Removing these lines was accidental. And I thought that I had corrected that. I will add them back in.

**Comment by sonofmun** [comment]
If you think it is OK to merge directly to master, then I will do that. There is no big hurry on this (it has already waited more than 2 months). Whatever you think is better.

**Comment by sonofmun** [comment]
A question, however: before these changes, were dev and master in sync? Because I used dev as the base branch for creating my own local `empty_reference` branch for these changes.

**Comment by sonofmun** [comment]
Oh and...mmmmm: American Breakfast!

**Comment by PonteIneptique** [comment]
Can you add a bumped version?

**Comment by PonteIneptique** [comment]
Dev is in advance over master.

**Comment by PonteIneptique** [comment]
Oh crap. That means master is gonna break.
Could you do the same changes from master instead that from dev?

**Comment by sonofmun** [comment]
But this PR still shows only 5 files (the same 5 files) being changed. But I would be more than happy to do it from master instead of dev.

**Comment by sonofmun** [comment]
NM, I see that there are more commits to these files. Will close this PR and open a new one.
